process.c

```diff
@@ -3020,77 +3005,67 @@ NORETURN(static VALUE f_exec(int c, const VALUE *a, VALUE _));
- *  otherwise, <code>ENV["RUBYSHELL"]</code> or <code>ENV["COMSPEC"]</code> on
- *  Windows and similar.  The command is passed as an argument to the
- *  <code>"-c"</code> switch to the shell, except in the case of +COMSPEC+.
+ *  The new process is created using the
+ *  {exec system call}[https://www.man7.org/linux/man-pages/man1/exec.1p.html];
+ *  it may inherit some of its environment from the calling program
+ *  (possibly including open file descriptors).
  *
- *  If the string from the first form (<code>exec("command")</code>) follows
- *  these simple rules:
+ *  Argument +env+, if given, is a hash that affects +ENV+ for the new process;
+ *  see {Execution Environment}[rdoc-ref:Process@Execution+Environment].
  *
- *  * no meta characters,
- *  * not starting with shell reserved word or special built-in,
+ *  Argument +options+ is a hash of options for the new process;
+ *  see {Execution Options}[rdoc-ref:Process@Execution+Options].
  *
- *  Ruby invokes the command directly without shell.
+ *  The first required argument is one of the following:
  *
- *  You can force shell invocation by adding ";" to the string (because ";" is
- *  a meta character).
+ *  - +command_line+ if it is a string,
+ *    and if it begins with a shell reserved word or special built-in,
+ *    or if it contains one or more metacharacters.
+ *  - +exe_path+ otherwise.
  *
- *  Note that this behavior is observable by pid obtained
- *  (return value of spawn() and IO#pid for IO.popen) is the pid of the invoked
- *  command, not shell.
+ *  <b>Argument +command_line+</b>
  *
- *  In the second form (<code>exec("command1", "arg1", ...)</code>), the first
- *  is taken as a command name and the rest are passed as parameters to command
- *  with no shell expansion.
+ *  \String argument +command_line+ is a command line to be passed to a shell;
+ *  it must begin with a shell reserved word or special built-in,
+ *  and may also contain arguments and options for that command.
  *
- *  In the third form (<code>exec(["command", "argv0"], "arg1", ...)</code>),
- *  starting a two-element array at the beginning of the command, the first
- *  element is the command to be executed, and the second argument is used as
- *  the <code>argv[0]</code> value, which may show up in process listings.
+ *  On a Unix-like system, the shell is <tt>/bin/sh</tt>;
+ *  otherwise the shell is determined by environment variable
+ *  <tt>ENV['RUBYSHELL']</tt>, if defined, or <tt>ENV['COMSPEC']</tt> otherwise.
  *
- *  In order to execute the command, one of the <code>exec(2)</code> system
- *  calls are used, so the running command may inherit some of the environment
- *  of the original program (including open file descriptors).
+ *  Except for the +COMSPEC+ case,
+ *  the entire string +command_line+ is passed as an argument
+ *  to {shell option -c}[https://www.man7.org/linux/man-pages/man1/bash.1.html#OPTIONS].
```

Comment:
Ruby uses `/bin/sh` not `bash`.
https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/utilities/sh.html

Reply:
Fixed.

Reply:
@BurdetteLamar I don't see this as fixed.